lib/rubocop/cop/lint/useless_return.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Lint
+      # This cop checks for the use of a return in a context where it will
+      # be ignored. (both in initialize and setter methods)
+      #
+      # @example
+      #
+      #   # bad
+      #   def initialize
+      #     foo
+      #     return :qux if bar?
+      #     baz
+      #   end
+      #
+      #   def foo=(bar)
+      #     return 42
+      #   end
+      #
+      class UselessReturn < Cop
+        include OnMethodDef
+
+        MSG = 'Do not use return in `%s`.'.freeze
+        def on_method_def(node, method_name, _args, _body)
+          return unless useless_return_method?(method_name) &&
+                        return_lines?(node)
+          add_offense(node, node.source_range, format(message, method_name))
+        end
+
+        private
+
+        def method_setter?(method_name)
+          method_name.to_s.end_with?('=') &&
+            !%i(!= == === >= <=).include?(method_name)
+        end
+
+        def return_lines?(node)
+          processed_source[line_range(node)].any? { |line| return_line?(line) }
```

Comment:
`return` has its own node type in the AST. No need to go searching in the raw source. 🙂